### PR TITLE
Module fmadm

### LINKED ISSUE #28781: [want] salt.module.fmadm
```
       The fmadm utility can be used by administrators and service personnel
       to view and modify system configuration parameters maintained by the
       Fault Manager, fmd(1M). fmd receives telemetry information relating to
       problems detected by the system software, diagnoses these problems, and
       initiates proactive self-healing activities such as disabling faulty
       components.
```

Would be awesome to have an exec module for this. Again I'll be happy to implement. (I will probably do so anyway)

Let's slowly make SmartOS and by extension in this case Solaris in some cases more of a first class minion!

### PR DESCRIPTION
**Pull request by sjorge**
Basics are there.

I'd like to implement fmadm.faulty verbose=True eventually... but none of the solaris servers at home or work are showing faults at the moment.

This closes #28781

### CODE REVIEW

File: salt/modules/solaris_fmadm.py [hunk 1]
```diff
@@ -0,0 +1,441 @@
+# -*- coding: utf-8 -*-
+'''
+Module for running fmadm and fmdump on Solaris
```
Comment (jfindlay) [inline salt/modules/solaris_fmadm.py]: @sjorge, `.. versionadded:: Boron` here.

File: salt/modules/solaris_fmadm.py [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+
+def __virtual__():
+    '''
+    Provides fmadm only on Solaris
+    '''
+    if salt.utils.is_sunos() and \
+        _check_fmadm() and _check_fmdump():
+        return __virtualname__
+    return False
```
Comment (jfindlay) [inline salt/modules/solaris_fmadm.py]: @sjorge, I would suggest here
```python
    return (False, '{0} module can only be loaded on Solaris with the fmadm'
                   ' and fmdump utilities installed'.format(__virtualname))
```
or similar.

**Comment by jfindlay** [comment]
@sjorge, thanks for your excellent work on this.

**Comment by sjorge** [comment]
I Will fix these before Monday.

**Comment by sjorge** [comment]
All fixed up, will go back to my other modules to include the better False return value too.

For (self) reference:
- salt versions -> https://docs.saltstack.com/en/latest/topics/releases/version_numbers.html

**Comment by cachedout** [comment]
Very clean code. Thanks, @sjorge

**Pull request merged**